### PR TITLE
 [backend/dwarfdbginf] finish removing MACHOBJ/ELFOBJ

### PR DESCRIPTION
**Pull request by thewilsonator**

### CODE REVIEW

File: src/dmd/backend/backconfig.d [hunk 1]
```diff
@@ -353,6 +354,10 @@ if (SYMDEB_CODEVIEW)
         cod3_set32();
     }
 
+    if (config.objfmt == OBJ_MACH)
+        machDebugSectionsInit();
+    else if (config.objfmt == OBJ_ELF)
+        elfDebugSectionsInit();
```
Comment (thewilsonator) [inline src/dmd/backend/backconfig.d:360]: I'm not sure where the best place to put this is

Comment (Luhrel) [reply src/dmd/backend/backconfig.d:360]: Put them in at the beginning of `_initfile`.
You may need to check if `debug_xxx == null`

Comment (thewilsonator) [reply src/dmd/backend/backconfig.d:360]: https://github.com/dlang/dmd/search?q=_initfile no results

Comment (Luhrel) [reply src/dmd/backend/backconfig.d:360]: https://github.com/dlang/dmd/blob/f99b447cbc986a0178059d5ff8cd7a2b1443051d/src/dmd/backend/dwarfdbginf.d#L1010

Comment (Luhrel) [reply src/dmd/backend/backconfig.d:360]: Or even better, in a `static this()`.

Comment (thewilsonator) [reply src/dmd/backend/backconfig.d:360]: > static this()

Won't work

Comment (thewilsonator) [reply src/dmd/backend/backconfig.d:360]: > https://github.com/dlang/dmd/blob/f99b447cbc986a0178059d5ff8cd7a2b1443051d/src/dmd/backend/dwarfdbginf.d#L1010

I thought I tried that. Will give it another shot

**Review by Luhrel — APPROVED** [review]
LGTM

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**